crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -625,35 +638,32 @@ impl<'src> Parser<'src> {
         let cls = match cls {
             Pattern::MatchAs(ast::PatternMatchAs {
                 name: Some(ident), ..
-            }) => {
-                if ident.is_valid() {
-                    Box::new(Expr::Name(ast::ExprName {
-                        range: ident.range(),
-                        id: ident.id,
-                        ctx: ExprContext::Load,
-                    }))
-                } else {
-                    #[allow(deprecated)]
-                    Box::new(Expr::Invalid(ast::ExprInvalid {
-                        value: self.src_text(&ident).into(),
-                        range: ident.range(),
-                    }))
+            }) => Box::new(Expr::Name(if ident.is_valid() {
+                ast::ExprName {
+                    range: ident.range(),
+                    id: ident.id,
+                    ctx: ExprContext::Load,
                 }
-            }
+            } else {
+                ast::ExprName {
+                    range: ident.range(),
+                    id: String::new(),
+                    ctx: ExprContext::Invalid,
+                }
+            })),
```

Comment:
Same as above. I don't think that converting any pattern to a name is the right thing to do here.